pandas/core/indexes/multi.py

```diff
@@ -3224,8 +3224,8 @@ def convert_indexer(start, stop, step, indexer=indexer, codes=level_codes):
                 return convert_indexer(start, stop + 1, step)
             else:
                 # sorted, so can return slice object -> view
-                i = level_codes.searchsorted(start, side="left")
-                j = level_codes.searchsorted(stop, side="right")
+                i = algos.searchsorted(level_codes, start, side="left")
```

Comment:
any idea why this is more performant?  wont algs.searchsorted end up calling level_codes.searchsorted anyway?

Reply:
if level_codes is of a different int type than start/stop, I suspect numpy.searchsorted ends up casting the entire level_codes array. algos.searchsorted attempts to cast the start/stop scalar only to match the type of level_codes before calling numpy.searchsorted.

Reply:
thanks